modules/localgov_services_status/src/Plugin/Block/ServiceStatusMessage.php

```diff
@@ -0,0 +1,142 @@
+<?php
```

Comment:
Is all of this needed to render the blocks? I wonder would it be possible to just create a view that lists all the statuses, and then place that block on "every" page. But since we are using conditional_fields module, the blocks will actually only appear on the pages they are needed.

I'm not sure if this is even possible to be honest, but it seems a more Drupally/site-builder way to do it.

Reply:
There's only one block and the code determines which status messages should be displayed based on the condition field. It doesn't look possible to use conditions in views so I'm not sure how this can be achieved just using views and without and custom code.

Reply:
So just to be clear in my mind.

Unlike alerts, which are entities, and which are shown in their own blocks (iirc) statuses are a node content type and shown will be shown in a single block. Statuses like alterts get the condition field which can be used to set visibility.

So this single block gets all the statuses content type items which are published. Then checks their visibility based on the conditions. Those left and then displayed in the block.
Even if it were a view, and a view filter based on visibility conditions, the view would have to do the same thing. Load the entities and then check the visibility via the Core Conditions Plugin API - as each condition plugin can pull in its own contexts (node type, route, user role etc.)